WooCommerce/UITestsFoundation/Screens/Menu/MenuScreen.swift

```diff
@@ -39,4 +47,16 @@ public final class MenuScreen: ScreenObject {
         app.buttons["dashboard-settings-button"].tap()
         return try SettingsScreen()
     }
+
+    @discardableResult
+    public func verifySelectedStoreDisplays(storeTitle expectedStoreTitle: String, storeURL expectedStoreUrl: String) -> MenuScreen {
+        let actualStoreTitle = selectedStoreTitleGetter(app).label
+        let actualStoreUrl = selectedStoreUrlGetter(app).label
+
+        XCTAssertEqual(expectedStoreTitle, actualStoreTitle,
+                       "Expected display name '\(expectedStoreTitle)' but '\(actualStoreTitle)' was displayed instead.")
```

Comment:
If I intentionally provide wrong actual value via `let actualStoreTitle = "Wrong actualStoreTitle"` this is the message I see:

`/Users/.../woocommerce-ios/WooCommerce/UITestsFoundation/Screens/Menu/MenuScreen.swift:56: error: -[WooCommerceUITests.LoginTests test_site_address_login_logout] : XCTAssertEqual failed: ("Your WooCommerce Store") is not equal to ("Wrong actualStoreTitle") - Expected display name 'Your WooCommerce Store' but 'Wrong actualStoreTitle' was displayed instead.`

As you can see, `XCTAssertEqual` still writes the default message with expected and actual values, even if custom message is provided.

Also note that the `verifySelectedStoreDisplays` method name is not mentioned anywhere, and at the same time your custom message does not clarify that the assertion is about store title/name. It just says "display name".

Because of this, I think the custom message should be used only if it provides more context than the default one. In this case, I think it's worth keeping the custom message if it's something like this:

```suggestion
        XCTAssertEqual(expectedStoreTitle, actualStoreTitle, "Store title inconsisteny in 'Menu'")
```

Or just not use it at all. WDYT?

(The same for the next assertion).

**UPD**: I later realized that this is the old code that you moved from the other screen... Still keeping my comment for history.

Reply:
Yup this is an existing assertion that was moved from another screen. But I see what you mean and agree with you that the custom message isn't very helpful in this case. I think it's better to not use it in this case since `XCTAssertEqual` already displays the mismatched values.

Updated here 82ac4a69e8c2bd7788fe3dcc8842efb60c86b477